ush/python/pygfs/jedi/jedi.py

```diff
@@ -0,0 +1,275 @@
+            #         config. Otherwise, if the algorithm J2-YAML is present, fill
+            #         its templates and merge.
+            if algorithm:
+                jcb_config['algorithm'] = algorithm
+            elif 'JCB_ALGO' in task_config.keys():
+                jcb_config['algorithm'] = task_config.JCB_ALGO
+            elif 'JCB_ALGO_YAML' in task_config.keys():
+                jcb_algo_config = parse_j2yaml(task_config.JCB_ALGO_YAML, task_config)
+                jcb_config.update(jcb_algo_config)
+
+            # Step 3: Generate the JEDI YAML using JCB
+            self.config = render(jcb_config)
+        elif 'JEDIYAML' in task_config.keys():
+            # Generate JEDI YAML without using JCB
+            self.config = parse_j2yaml(task_config.JEDIYAML, task_config,
+                                       searchpath=self.j2tmpl_dir)
+        else:
+            raise KeyError(f"Task config must contain JCB_BASE_YAML or JEDIYAML")
+
+    @logit(logger)
+    def execute(self, task_config: AttrDict, aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
+        """Execute JEDI application
+
+        Parameters
+        ----------
+        task_config: AttrDict
+            Attribute-dictionary of all configuration variables associated with a GDAS task.
+        aprun_cmd: str
+            String comprising the run command for the JEDI executable.
+        jedi_args (optional): List
+            List of strings comprising optional input arguments for the JEDI executable.
+
+        Returns
+        ----------
+        jedi_config: AttrDict
+            Attribute-dictionary of JEDI configuration rendered from a template.
+        """
+
+        chdir(task_config.DATA)
+
+        exec_cmd = Executable(aprun_cmd)
+        exec_cmd.add_default_arg(self.exe)
+        if jedi_args:
+            for arg in jedi_args:
+                exec_cmd.add_default_arg(arg)
+        exec_cmd.add_default_arg(self.yaml)
+
+        try:
+            exec_cmd()
+        except OSError:
+            raise OSError(f"Failed to execute {exec_cmd}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
+
+    @logit(logger)
+    def link_exe(self, task_config: AttrDict) -> None:
+        """Link JEDI executable to run directory
+
+        Parameters
+        ----------
+        task_config: AttrDict
+            Attribute-dictionary of all configuration variables associated with a GDAS task.
+
+        Returns
+        ----------
+        None
+        """
+
+        # TODO: linking is not permitted per EE2.
+        # Needs work in JEDI to be able to copy the exec. [NOAA-EMC/GDASApp#1254]
+        logger.warn("Linking is not permitted per EE2.")
+        exe_dest = os.path.join(task_config.DATA, os.path.basename(task_config.JEDIEXE))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(task_config.JEDIEXE, exe_dest)
```

Comment:
```suggestion
    @staticmethod
    @logit(logger)
    def link_exe(task_config: AttrDict) -> None:
        """Link JEDI executable to run directory

        Parameters
        ----------
        task_config: AttrDict
            Attribute-dictionary of all configuration variables associated with a GDAS task.

        Returns
        ----------
        None
        """

        # TODO: linking is not permitted per EE2.
        # Needs work in JEDI to be able to copy the exec. [NOAA-EMC/GDASApp#1254]
        logger.warn("Linking is not permitted per EE2.")
        exe_dest = os.path.join(task_config.DATA, os.path.basename(task_config.JEDIEXE))
        if os.path.exists(exe_dest):
            rm_p(exe_dest)
        os.symlink(task_config.JEDIEXE, exe_dest)
```
This is a `staticmethod` as it makes no use of `self`.